### ISSUE TITLE
Table Editor - Text shows up only after clicking somewhere else - Windows & Linux - Zim Version 0.74.1

### ISSUE THREAD

**Issue by fdt-dev** [zim-desktop-wiki/zim-desktop-wiki#1677]
**Describe the bug**
Zim Version 0.74.1 for Windows and Linux.
Text entered into the table cells does not show up while typing. No cursor shows up. Text shows up when user is clicking someplace else. Very erratic, not functional.
Table editing currently works better by directly editing the Zim article via an external editor. 

**Expected behavior**
Behaviour of the table editor should be like the pre 0.74.x releases. Text cursor should show up in the cells, entered text should show up immediately.

**Environment (please complete the following information):**
 - Version 0.74.1 
 - OS: Linux/Windows (possibly Mac; not tested, not available)
 - Language: German

**Comment by bjjanssen** [comment]
Can confirm. Unfortunatly I moved back to 0.73.5 before taking a look at diagnostics, but I needed stuff to be done.

-     Version 0.74.2
-     OS: MS Windows 10
-     Language: German

There seem to be two kinds of wrong behavior: in text mode, and in table extension mode. 
I usually don't use the table extension, so my experience there is limited.

In text mode: 
Redrawing/aligning of table does not happen on reload (CTRL+r). 
Toggling "editable/not editable" redraws table once.
Adding or removing | (pipes) AND/OR new lines sometimes produces a line like this
`				  |          |               |      |      |      |      |      |      |   }}}`
Removing the above breaks the template and shows a tag at the beginning of the table like this
`{{{ table:                                              `
Once the tags show up, copy and paste ceases to function properly. 
Cursor turns invisible, not only in the table but in the whole document.

In extension mode:
Enabling/disabling the extension redraws the table once.
Cursor becomes invisible in the table.



Attempted solutions:
Disabling dark mode -- I thought maybe the cursor does not show up because of a color choice. 
Adding and removing the table extension -- several times ;)
Rebuilding the table in external editor and copy & pasting it into Zim. Works one time, then familiar breakage.

**Comment by jaap-karssenberg** [comment]
Did see this issue happen as well myself, but cannot reproduce it now. So need to investigate how / where to reproduce. Might actually be an upstream issue.

**Comment by bjjanssen** [comment]
Some more info:

OS: GNU/Linux (Arch)
Version: 0.74.3
Language: English

Same behaviour. However, I'm syncing my Zim Notebooks to and fro the MS Windows (version 11, Zim version 0.73.5, German).

If I create a page in Zim 0.73.5 and add a table, everything is fine. Sync it and edit the table on Linux with Zim 0.74.3, it breaks like described. Sync it again and the breakage is also there in 0.73.5.

When I look into the text file, I see lines such as 
`{ { { table:`
or 
`| | | | | |} } }`
or lots of empty space and an abundance of pipes in the middle of the table.

When I clean up the file by hand, it is displayed correctly and editable in 0.73.5 again. 0.74.3 borks it up again.

BTW, is there a way to get rid of the "You need extension X to display this object" frame? I don't need or want the extension, I just want the auto-formatting.

**Comment by aloboa** [comment]
I have this issue on linux with zim 0.74.3 I cannot visualize what I write on a cell until I click outside.
Oddly, if I close zim and reopen, editing the table becomes normal (the cell becomes highlitghted in blue and I can see what I write).

**Comment by hsandt** [comment]
Confirmed on Zim 0.75.2 on Linux Ubuntu 22.04 with Unity desktop.

![2023-07-13 Table caret and live input not visible until confirm](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/28d4ffce-b3a9-4d34-9702-d3113c40ba9b)

(note: cursor does

Happens even when table is filled. Also, I still need to click twice on each cell as noted in https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/591, or the new text will appear at last caret position outside the table.

**Comment by scubbx** [comment]
Linux Mint XFCE, ~~FlatPack~~ _(edit)_ DEB version of ZIM 0.75.2
I can confirm exactly the same behaviour. Entering or editin Text is only possible by double-clicking the cell first. Durint text-input, the changes are not displayed. Only after pressing [ENTER] the changes are visually applied.

![table_problem](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1347421/4b5bf035-d828-4267-a447-266aab38f2fb)

_(edit)_ Switching to the FlatPack version did solve the Problem.

**Comment by Yorizuka** [comment]
Void linux, same issue

**Comment by Yorizuka** [comment]
Would it be possible to add a "edit as raw text" feature, this would be nice to have regardless, but would also help work around this issue until its fixed properly.

**Comment by Skurtov** [comment]
Have same issue. Very inconvenient that the table editor is essentially non-functional even with how limited it is in general bug is still present on newest build - 0.76.3 on Windows 11.
